flang/lib/Lower/Bridge.cpp

```diff
@@ -1473,7 +1473,26 @@ class FirConverter : public Fortran::lower::AbstractConverter {
   /// the global to the ModuleOp as a new uniqued symbol and initialize it with
   /// the correct value. It will be referenced on demand using `fir.addr_of`.
   void instantiateGlobal(const Fortran::lower::pft::Variable &var) {
-    llvm_unreachable("Varun: put your code here");
+    const auto &sym = var.getSymbol();
+    std::string globalName = mangleName(sym);
+    fir::GlobalOp global;
+    if (builder->getNamedGlobal(globalName))
+      return;
+    if (const auto *details =
+            sym.detailsIf<Fortran::semantics::ObjectEntityDetails>()) {
+      if (details->init())
+        global = builder->createGlobal(
+            toLocation(), genType(sym), globalName, false,
+            [&](Fortran::lower::FirOpBuilder &builder) {
+              auto initVal = genExprValue(details->init().value());
+              builder.create<fir::HasValueOp>(toLocation(), initVal);
+            });
+      else
+        global = builder->createGlobal(toLocation(), genType(sym), globalName);
+      auto addrOf = builder->create<fir::AddrOfOp>(
+          toLocation(), global.resultType(), global.getSymbol());
+      addSymbol(sym, addrOf);
+    }
```

Comment:
Please put an `} else { TODO();` here so that we catch any missing functionality.

Reply:
If a symbol doesn't have `Fortran::semantics::ObjectEntityDetails` it should not be showing up here.
So I think I should have a `} else { llvm_unreachable(""); }`. What do you think?